DependencyInjection/DoctrineExtension.php

```diff
@@ -343,6 +345,11 @@ protected function ormLoad(array $config, ContainerBuilder $container)
         $loader = new XmlFileLoader($container, new FileLocator(__DIR__ . '/../Resources/config'));
         $loader->load('orm.xml');
 
+        // This check can be removed when support for Symfony < 4.2 is dropped
+        if (! interface_exists(ResetInterface::class) || ! is_a(DoctrineType::class, ResetInterface::class)) {
```

Comment:
Shouldn't it be the other way around (i.e. `interface_exists(ResetInterface::class) || is_a(DoctrineType::class, ResetInterface::class)`)?

Reply:
We should add this service tag only in case when `DoctrineType` does not implement `ResetInterface`. But before this we need to check whether `ResetInterface` exists, and if not - we have old Symfony and also need to add this tag.

Or do you propose change condition from `(! a || ! b)` to `(! (a && b))` ?

Reply:
We need to add the tag ourselves if any of the following conditions are true:
* there is no ResetInterface
* the type doesn’t implement ResetInterface
* FrameworkBundle does not have autoconfiguration for ResetInterface (checked by method_exists on the controller)

Reply:
we should not rely on autoconfiguration IMO, and always add the tag explicitly when we need to have it.

Reply:
if we only support versions defining the interface in the DoctrineType, we should put the tag directly in XML.